src/components/HomePage/index.js

```diff
@@ -1,27 +1,48 @@
-import React, { useEffect, useState } from 'react';
-import { Link } from 'react-router-dom'; 
-import BookCard from '../BookCard';
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import BookCard from "../BookCard";
+import GenreFilter from "../GenreFilter";
+import "./home.css";
+import SearchBar from "../SearchBar";
 
 function HomePage() {
-    const [books, setBooks] = useState([]);
+  const [books, setBooks] = useState([]);
+  const [genreID, setGenreID] = useState(0);
+  const [search, setSearch] = useState("");
 
-    useEffect(() => {
-        fetch('https://book-swap-api.dev.io-academy.uk/api/books?claimed=0')
-            .then(res => res.json())
-            .then(bookData => {
-                setBooks(bookData.data);
-            });
-    }, []);
+  useEffect(() => {
+    let url = "https://book-swap-api.dev.io-academy.uk/api/books?claimed=0";
 
-    return (
-        <div className='flex_container'>
-            {books.map((book, index) => (
-                <Link key={index} to={`/Book/${book.id}`}>
-                    <BookCard book={book} />
-                </Link>
-            ))}
-        </div>
-    );
+    if (genreID > 0) {
+      url += `&genre=${genreID}`;
+    }
+
+    if (search.length > 0) {
+      url += `&search=${search}`;
+    }
+
+    fetch(url)
+      .then((res) => res.json())
+      .then((bookData) => {
+        setBooks(bookData.data);
+      });
+  }, [genreID, search]);
+
+  return (
+    <div>
+      <div className="search-container">
+        <GenreFilter genreID={genreID} setGenreID={setGenreID} />
+        <SearchBar search={search} setSearch={setSearch} />
+      </div>
+      <div className="flex_container">
+        {books.map((book, index) => (
```

Comment:
Worth now thinking about what happens if there are no books, which is possible with claimed/unclaimed. Just displaying a simple message if books is empty would do.

Reply:
This comment still stands. Show a basic 'no books found' message if books is empty